src/ngAria/aria.js

```diff
@@ -377,6 +377,27 @@ ngAriaModule.directive('ngShow', ['$aria', function($aria) {
             });
           }
         }
+
+        if ($aria.config('bindKeypress') && isNodeOneOf(elem, ['A', 'BUTTON'])) {
+          elem.on('keypress', function(event) {
+            var keyCode = event.which || event.keyCode;
+
+            if (elem[0].nodeName === 'A') {
+              var hasHref = elem.attr('href') != null && elem.attr('href') !== '';
```

Comment:
You just have to check if `elem.attr('href')` is falsy.

BTW, if we need to also support SVG, then we need to check `xlink:href` (instead of `href`).
(You can take a look at the `htmlAnchorDirective` in the source code for more details.)